diff --git a/homework1/train.py b/homework1/train.py
index d47080f..51712c6 100644
--- a/homework1/train.py
+++ b/homework1/train.py
@@ -24,7 +24,7 @@ parser.add_argument("--epochs",type=int,nargs="?",default = 100,help="train epoc
 parser.add_argument("--alias",type=str,nargs="?",default=0,
                     help = "Please enter the alias")
 parser.add_argument("--wandb",action="store_false")
-parser.add_argument("--batch-size","-bs",type=int,nargs="?",default = 64,
+parser.add_argument("--batch-size","-bs",type=int,nargs="?",default = 32,
                     help = "train batch size")
 parser.add_argument("--learning-rate","-lr",nargs="?",default=1e-3,
                     help = "learning rate")
