diff --git a/homework1/train.py b/homework1/train.py
index d47080f..4bcafbd 100644
--- a/homework1/train.py
+++ b/homework1/train.py
@@ -30,7 +30,7 @@ parser.add_argument("--learning-rate","-lr",nargs="?",default=1e-3,
                     help = "learning rate")
 parser.add_argument("--optimer",type=str,nargs="?",default="adam",
                     help = "optimier to train")
-parser.add_argument("--hidden-dim",type=int,nargs="?",default = 32,
+parser.add_argument("--hidden-dim",type=int,nargs="?",default = 8,
                     help = "width size of the NN")
 parser.add_argument("--n-layers",type=int,nargs="?",default=3,
                     help = "depth size of the NN")
