diff --git a/homework1/train.py b/homework1/train.py
index d47080f..7734c0e 100644
--- a/homework1/train.py
+++ b/homework1/train.py
@@ -32,7 +32,7 @@ parser.add_argument("--optimer",type=str,nargs="?",default="adam",
                     help = "optimier to train")
 parser.add_argument("--hidden-dim",type=int,nargs="?",default = 32,
                     help = "width size of the NN")
-parser.add_argument("--n-layers",type=int,nargs="?",default=3,
+parser.add_argument("--n-layers",type=int,nargs="?",default=4,
                     help = "depth size of the NN")
 parser.add_argument("--activation",type = str,nargs="?",default = "Relu",
                     help = "nonlinear activation function")
